markriverac/academy-go-q42021
# Deliverable 2

# Second Deliverable (due November 26th 23:59PM)
## Based on the self-study material and mentorship covered until this deliverable, we suggest you perform the following:

1. Create a client to consume an external API
1. Add an endpoint to consume the external API client
1. The information obtained should be stored in the CSV file
1. Add unit testing
1. Update the endpoint made in the first deliverable to display the result as a JSON
1. Refator if needed

Note: what’s listed in this deliverable is just for guidance and to help you distribute your workload; you can deliver more or fewer items if necessary. However, if you deliver fewer items at this point, you have to cover the remaining tasks in the next deliverable.

## How to test it

1. Move to the `main` package.
2. Run `go run main.go`.
3. Running this command will call to the public Pokemon API.
4. It should create a `pokemon.csv` file.
5. It should load the. `.csv` file with the first 150 pokemon in the pokedex. You can modify this param in the `main.go` function.
3. On your explorer go to `http://localhost:10000/`. A message indicating the server is running should appear.
3. On your explorer go to `http://localhost:10000/pokemon`. A list of all items in the `pokemon/pokemon.csv` file should appear as a JSON.
5. Go to `http://localhost:10000/pokemon/{name}` where `name` is any name of a pokemon in the CSV file. A JSON with the matching item should be visible.
6. Repeat step 5 but use an id that doesn't exist. A message indicating that the item doesn't exist should appear.
